### ISSUE TITLE
Medical Transformer: Gated Axial-Attention for Medical Image Segmentation

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#20]
## Abstract
* 기존의 deep convolutional neural networks는 medical image segmentation에 있어 understanding of **_long-range dependencies_** in the image가 부족하다.
* self-attention mechanism은 long-range dependencies를 encode한다.
* 일반적인 vision application을 위한 datasets과 달리 medical 기반은 data가 크지 않다.
* a gated axial-attention model which extends the existing architectures by introducing an additional control mechanism in the self-attention module을 제안한다.
* 거기에 적은 양의 medical images를 효과적으로 학습시키기 위해 Local-Global training strategy(LoGo)를 제안한다.
* global features에는 whole image로 넣고, local features에는 patches로 넣는다.
* Medical Transformer(MedT) 모델은 3가지의 medical image segmentation datasets으로 평가한다.

## Introduction
1. proposes a gated position-sensitive axial attention mechanism that works well even on smaller datasets
2. introduces Local-Global (LoGo) training methodology for transformers which is effective
3. proposes Medical-Transformer (MedT) which is built upon the above two concepts proposed specifically for medical image segmentation
4. successfully improves the performance for medical image segmentation tasks over convolutional networks and fully attention architectures on three different datasets

## MedT 모델 설명

* 전체적인 모델
<image src="https://user-images.githubusercontent.com/67627471/128007531-2755d151-9121-46ba-9206-3649649ef3fc.png" width="80%" height="80%" />

##
* 아래 각각의 식은 width-wise만 나타나 있지만 아래 첨자 w만 바꾼 동일한 식으로 height axis에도 적용함!!

* 처음 self-attention 식(점차 업그레이드 됨)
<image src="https://user-images.githubusercontent.com/67627471/128007786-3be2ca49-7913-4eb9-896c-21879ee1e2f7.png" width="50%" height="50%" />

> the values v are pooled based on global affinities calculated using softmax
  지금 이 식만으로는 아래 두 가지 문제점이 있음.
>  * computationally very expensive and with increased feature map size it often becomes infeasible to use self-attention for vision model architectures
>  * unlike convolutional layer, self-attention layer does not utilize any positional
information while computing the non-local context.

##
* axial-attention을 더한 식
<image src="https://user-images.githubusercontent.com/67627471/128007814-2d3f8b96-d6ba-453f-b538-ddbdddf472bc.png" width="50%" height="50%" />

> relative positional encoding이라고도 하는 positional bias embedding을 적용.
  위의 식에서 r부분이 positional 부분임.
  affinities가 positional information에 sensitive해지도록 함.

##
* gating mechanism을 추가한 식
<image src="https://user-images.githubusercontent.com/67627471/128007728-bb3429e6-d63d-4bd5-8f0a-01b5c1f31716.png" width="60%" height="60%" />

> relative positional encoding이 정확하게 학습되지 않았으면 낮은 가중치를 갖고, 정확하게 학습된 것들이 높은 가중치를 갖게 해줌.

## 결과
<image src="https://user-images.githubusercontent.com/67627471/128009830-06a04170-7ef8-4057-b31b-8ef113d6808d.png" width="90%" height="90%" />

* (e)가 전문의에 의해 segment된 ground truth이고, (d)가 본 논문에서 제안한 MedT의 결과.
* image에 까만 펜으로 색칠된 건(제가 칠한 부분) 많은 모델이 구조물로 오해한 부분.

<image src="https://user-images.githubusercontent.com/67627471/128010102-a127b310-5459-4dbf-9ae2-90d783604d4a.png" width="80%" height="80%" />

* Gated Axial self-attention과 LoGo를 모두 적용한 MedT의 성능이 가장 좋은 걸 알 수 있음.

##
* 이 논문 매우 흥미롭습니다! 혹시 transformer에 관심 있으시다면 medical 쪽 연구 안 하셔도 유익할 것 같아요!
* long range에 있는 요소 간의 attention을 가능하게 해서 segmentation의 정확도를 높인 모델.
* 많은 양의 data가 필요한 transformer를 적은 양의 medical image에 적용할 수 있도록 local/global 나눠서 2배로 돌린 효과 good.
